Core/src/org/sleuthkit/autopsy/casemodule/MultiUserCase.java

```diff
@@ -0,0 +1,285 @@
+     *
+     * @return The case display name.
+     */
+    String getCaseDisplayName() {
+        return this.caseDisplayName;
+    }
+
+    /**
+     * Gets the creation date for the case, defined as the create time of the
+     * case metadata file.
+     *
+     * @return The case creation date.
+     */
+    Date getCreationDate() {
+        return this.createDate;
+    }
+
+    /**
+     * Gets the last accessed date for the case, defined as the last accessed
+     * time of the case metadata file.
+     *
+     * @return The last accessed date.
+     */
+    Date getLastAccessedDate() {
+        return this.lastAccessedDate;
+    }
+    
+    /**
+     * Gets metadata (.aut) file name.
+     * 
+     * @return The metadata file name.
+     */
+    String getMetadataFileName() {
+        return this.metadataFileName;
+    }
+
+    /**
+     * Gets the status of this case based on the auto ingest result file in the
+     * case directory.
+     *
+     * @return See CaseStatus enum definition.
+     */
+    CaseStatus getStatus() {
+        if(caseDirectoryPath.resolve("autoingest.alert").toFile().exists()) {
+            return CaseStatus.ALERT;
+        } else {
+            return CaseStatus.OK;
+        }
+    }
+
+    /**
+     * Gets the case metadata from a case directory path.
+     *
+     * @param caseDirectoryPath The case directory path.
+     *
+     * @return Case metadata.
+     * 
+     * @throws CaseMetadata.CaseMetadataException If the CaseMetadata object
+     *                                            cannot be constructed.
+     * @throws MultiUserCaseException             If no case metadata (.aut)
+     *                                            file is found in the case
+     *                                            directory.
+     */
+    private static CaseMetadata getCaseMetadataFromCaseDirectoryPath(Path caseDirectoryPath) throws CaseMetadata.CaseMetadataException, MultiUserCaseException {
+        CaseMetadata caseMetadata = null;
+        
+        File directory = new File(caseDirectoryPath.toString());
+        if (directory.isDirectory()) {
+            File autFile = null;
+            
+            /*
+             * Attempt to find an AUT file via a directory scan.
+             */
+            for (File file : directory.listFiles()) {
+                if (file.getName().toLowerCase().endsWith(CaseMetadata.getFileExtension()) && file.isFile()) {
+                    autFile = file;
+                    break;
+                }
+            }
+            
+            if(autFile == null || !autFile.isFile()) {
+                throw new MultiUserCaseException(String.format("No case metadata (.aut) file found in the case directory '%s'.", caseDirectoryPath.toString()));
+            }
+            
+            caseMetadata = new CaseMetadata(Paths.get(autFile.getAbsolutePath()));
+        }
+        
+        return caseMetadata;
```

Comment:
@dgrove727 , update: while the MultiUserCaseManager is looking for the case metadata file, it might as well also look for the case auto ingest log.

Reply:
The way it works right now is upon clicking on a row in the table of cases, it will check for the existence of the log file for that single case based on the case directory path. Are you saying you want me to grab the log path for each case we iterate through and store it in each object?

Reply:
Yes, that is what I'm saying. I think there is a tiny, tiny chance that the file could appear in the interval after the MultiUserCase object is created, but I can live with that.